### PR TITLE
Feature publisher cru + query

### PR DESCRIPTION
**Pull request by aatuny**
- Fixed publisher create operation
- Added publisher read + query operation with support to different levels of information returned based on user role
- Disabled delete endpoints for publisher and publisher request entities
- Updated apidoc for publisher request and publisher endpoints

### CODE REVIEW

File: src/interfaces/requestsPublishers.js [hunk 1]
```diff
@@ -80,8 +79,8 @@ export default function () {
 
   async function updateRequest(db, id, doc, user) {
     try {
-      if (validateDoc(doc, 'PublisherRequestContent')) {
-        if (hasPermission(user, 'publisherRequests', 'updateRequest')) {
+      if (hasPermission(user, 'publisherRequests', 'updateRequest')) {
+        if (validateDoc(doc, 'PublisherRequestContent')) {
           // Set backgroundProcessingState based on earlier if not re-defined in request
           const {backgroundProcessingState} = await publisherRequestsInterface.read(db, id, {});
           const newDoc = {...doc, backgroundProcessingState: doc.backgroundProcessingState ? doc.backgroundProcessingState : backgroundProcessingState};
```
Comment (mvillivald) [inline src/interfaces/requestsPublishers.js:86]: Can it be rewritten like this?
```suggestion
          const newDoc = {...doc, backgroundProcessingState: doc.backgroundProcessingState || backgroundProcessingState};
```

File: test-fixtures/publishers/update/2/metadata.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+    "descr": "Should succeed, admin can update publishers",
+    "requestUrl": "/publishers/111111111111111111111111",
+    "username": "system",
+    "password": "system",
```
Comment (mvillivald) [inline test-fixtures/publishers/update/2/metadata.json:5]: Shouldn't it be:
```suggestion
    "username": "admin",
    "password": "admin",
```

**Review by mvillivald — APPROVED** [review]
There are a couple of comments, however, I do not believe those things affect functionality in any way, so otherwise, everything is fine.

File: src/utils.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ export function mapRoleToGroup(role) {
 export function mapGroupToRole(group) {
   const data = GROUPS_AND_ROLES;
   const value = Object.values(data).find(v => group.includes(v));
-  return Object.keys(data).find(key => data[key] === value);
+  return value ? Object.keys(data).find(key => data[key] === value) : undefined;
```
[the remark this answers is not in the document]
Comment (aatuny) [reply src/utils.js:60]: Theoretically yes it should be, however preferring to keep styling consistent here.

File: src/interfaces/requestsPublishers.js (hunk 1 above)
Comment (aatuny) [reply src/interfaces/requestsPublishers.js:86]: Should be ok, but keeping the coding style consistent with previously written API styling.

**Pull request merged**